innovation.game.php

```diff
@@ -11727,8 +11836,38 @@ function stInterInteractionStep() {
                         }
                     }
                     break;
-                }
 
+                // id 155, Artifacts age 5: Boerhavve Silver Microscope
+                case "155N1B":
+                    // "Draw and score a card of value equal to the sum of the values of the cards returned."
+                    $first_age = self::getGameStateValue('auxiliary_value');
+                    $second_age = self::getGameStateValue('age_last_selected');
+                    self::executeDraw($player_id, $first_age + $second_age, 'score');
+                   
+                    break;
+                    
+                // id 156, Artifacts age 5: Principia
+                case "156N1A":
+                    $ages_on_top = self::getGameStateValueAsArray('auxiliary_value');
+                    sort($ages_on_top);
+                    // For each card returned, draw and meld a card of value one higher than the value of the returned card, in ascending order.
+                    foreach($ages_on_top as $card_age) {
+                        self::executeDraw($player_id, $card_age + 1, 'board');
+                    }
+                    break;
+
+                // id 160, Artifacts age 5: Hudson's Bay Company Archives
+                case "160N1A":
+                    $color = self::getGameStateValue('color_last_selected');
+                    $board = self::getCardsInLocation($player_id, 'board', null, false, true);
+                    $pile = $board[$color];
+                    if (count($pile) > 1) {
```

Comment:
There's no need to check the count of the pile, since this is done inside the splay method.

Reply:
ah ok.  its hard to know what I should check for...heh

Reply:
will do.